### ISSUE TITLE
tf.keras is ignoring specified step_per_epoch when keras doesn't

### ISSUE THREAD

**Issue by lonelykid** [tensorflow/tensorflow#27353]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):no
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):unknown
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):colab
- TensorFlow version (use command below):1.13
- Python version:python 3
- Bazel version (if compiling from source):unknown
- GCC/Compiler version (if compiling from source):unknown
- CUDA/cuDNN version:unknown
- GPU model and memory:unknown

You can collect some of this information using our environment capture [script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with
python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

**Describe the current behavior**
`history = model.fit_generator(
      train_generator,
      steps_per_epoch=8,  
      epochs=15,
      verbose=1)`
If train_generator is a instance of Sequence.( in my case, coming from flow_from_directory), steps_per_epoch is overridden by len(train_generator) in this file [training_generator.py](https://github.com/tensorflow/tensorflow/blob/r1.13/tensorflow/python/keras/engine/training_generator.py#L371)
This doesn't happen in keras. In keras, steps_per_epoch is kept.

**Describe the expected behavior**
Use specified steps_per_epoch instead of len(data) if specified.

**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.
I find this when doing a coursera online course. The notebook i used can be found at [link](https://github.com/lmoroney/dlaicourse/blob/master/Course%201%20-%20Part%208%20-%20Lesson%202%20-%20Notebook.ipynb)

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by jvishnuvardhan** [comment]
This is not Build/Installation or Bug/Performance issue. Please post this kind of support questions at [Stackoverflow](https://stackoverflow.com/questions/tagged/tensorflow). There is a big community to support and learn from your questions. GitHub is mainly for addressing bugs in installation and performance. Thanks!

**Comment by lonelykid** [comment]
@jvishnuvardhan I understand what is happening. I don't need stackoverflown for clarification. 
The problem I'm reporting is that the parameter is not working as expected in tf.keras, and is behaving differently in keras and tf.keras. I heard that you are trying to keep tf.keras api in agreement with keras. This could be fixed on either ends to make them behave the same.

**Comment by hyzhak** [comment]
got the same issue with the last keras and tensorflow:

```
tf.__version__, tf.keras.__version__
('1.13.1', '2.2.4-tf')
```

**Comment by omalleyt12** [comment]
Thanks for the issue! It seems reasonable to prefer `steps_per_epoch` (if provided) over the `Sequence` length, will look into this

**Comment by omalleyt12** [comment]
Looks like this has already been fixed in the TF 2.0 nightly
